addons/modules/functions/fnc_moduleLocation.sqf

```diff
@@ -0,0 +1,86 @@
+private _sides = [
+    east,
+    west,
+    independent,
+    civilian,
+    sideUnknown
+];
+
+private _attributes = [
+    ["COMBO", "str_3den_trigger_attribute_type_displayname", [_types,_types, _types find _type]],
+    ["EDIT", "str_a3_cfgvehicles_modulecuratoraddicon_f_arguments_text", _text, true],
+    ["SLIDER", "STR_3DEN_Display3DEN_ControlsHint_Rotate", [0, 360, _direction, EFUNC(common,formatDegrees)], true],
+    ["TOOLBOX", "str_3den_trigger_attribute_shape_tooltip", [parseNumber _isRectangular, 1, 2, ["str_3den_attributes_shapetrigger_ellipse", "str_3den_attributes_shapetrigger_rectangle"]]],
+    ["EDIT", format ["%1 %2", localize "str_3den_trigger_attribute_size_displayname",localize "str_3den_axis_x"], _size select 0, true],
+    ["EDIT", format ["%1 %2", localize "str_3den_trigger_attribute_size_displayname",localize "str_3den_axis_y"], _size select 1, true],
+    ["EDIT", "STR_3DEN_Display3DENSave_Filter_Name_text", _name, true],
+    ["COMBO", "str_a3_rscdisplaydynamicgroups_side", [_sides, [
+        "OPFOR",
+        "BLUFOR",
+        "Independent",
+        "Civilian",
+        "Unknown"
+    ], _sides find _side]],
+    ["EDIT", LSTRING(ModuleLocation_Importance), _importance, true]
+];
+
+["STR_3DEN_Display3DEN_Locations", _attributes, {
+    params ["_dialogValues", "_location"];
+    _dialogValues params [
+        "_type",
+        "_text",
+        "_direction",
+        "_isRectangular",
+        "_sizeX",
+        "_sizeY",
+        "_name",
+        "_side",
+        "_importance"
+    ];
+
+    private _location = createLocation [_location];
+
+    _location setType _type;
+    _location setText _text;
+    _location setDirection _direction;
+    _location setRectangular ([false, true] select _isRectangular);
```

Comment:
Toolbox controls can return `BOOL` if default is supplied as `BOOL` and there are only two options.